server/src/main/java/org/opensearch/indices/replication/SegmentReplicationTarget.java

```diff
@@ -169,8 +167,7 @@ private void getFiles(CheckpointInfoResponse checkpointInfo, StepListener<GetSeg
                 )
             );
         }
-        final List<StoreFileMetadata> filesToFetch = Stream.concat(diff.missing.stream(), diff.different.stream())
-            .collect(Collectors.toList());
+        final List<StoreFileMetadata> filesToFetch = diff.missing.stream().collect(Collectors.toList());
```

Comment:
minor - diff.missing is already a list, so we shouldn't need to stream & collect to a new list.